### ISSUE TITLE
Jumping to line with error after typing "e" in MiKTeX error prompt

### ISSUE THREAD

**Issue by francoisglineur** [TeXworks/texworks#965]
**Bug description:**
I am using the latest TeXworks (0.6.7) with MiKTeX 2.9 on Windows 10. When compiling, in case there is an error, typing "e" in the MiKTeX error prompt should open the TeXworks editor at the line where the error occurs, but nothing happens.

**Steps to reproduce the problem:**

1. Configure MiKTeX to use the Texworks editor to deal with errors, by setting the environment variable MIKTEX_EDITOR to the following: "C:\Program Files (x86)\TeXworks\texworks.exe" --position=+%l "%f"
2. Compile a LaTeX document containing some error (with pdflatex).
3. Wait until the MiKTeX compiler stops and type "e" + enter at the error prompt
4. Nothing happens

**Expected behavior:**
The texworks editor should jump to the line where the error occurs.

**General information:**
_TeXworks version_: 0.6.7
_TeXworks obtained from_: https://github.com/TeXworks/texworks/releases
_Operating system_: Windows 10

**Additional information:**
I find that pressing "e" is very helpful when debugging LaTeX documents (faster than navigating to the "Errors, warnings, ..." pane and clicking on the error).
I know that MiKTeX is configured correctly because when compiling the same file on the command line, typing "e" does open Texworks at the correct line. Adding "-shell-escape" does not solve the problem. Apologies in advance if this is not a texworks bug but a misconfiguration on my side.

**Comment by stloeffler** [comment]
There are a few interesting things about this:
First, why do you have the plus sign in `-position=+%l` - or was this just a typo?
Second, TeXworks automatically adds (or overrides) the MIKTEX_EDITOR environment variable when starting a typesetting process (see [Engine.cpp](https://github.com/TeXworks/texworks/blob/a339582fb5aeea7157c7ae7ce0e7146e79003844/src/Engine.cpp#L131)). Whether or not it should do this (override a variable set by the user elsewhere) is another topic for debate, but in my tests it did. One relatively easy way to confirm this is to create a new typesetting tool (e.g. named `env`) which runs `cmd` with the two arguments `/c` and `set`, which prints all environment variables passed to the typesetting tool.
Third, in my tests, TeXworks did successfully jump to the right location, though this was a minimal example (no multi-file thesis-length document) with old versions of MiKTeX (2.9) and Windows (8.1).

**Comment by francoisglineur** [comment]
Thanks a lot for the quick reply! About your first three points:
1. The + sign was actually present in my MIKTEX_EDITOR variable ; it was copied/adapted from https://tex.stackexchange.com/questions/579059/latex-compiller-and-editor-interaction
(apparently things work both with and without it)
2. This is very useful information (I would have had a hard time guessing that). I could confirm this behaviour with a batch file too. The path I see in MIKTEX_EDITOR is: C:/Program Files (x86)/TeXworks/TeXworks.exe --position=%l "%f"
3. In my tests with TeXworks the jump was not working. However, seeing the path above hints at two possible problems: the path separator "/" which is in the wrong direction for Windows, and the spaces present in the path. 
Hence I tried moving TeXworks to another directory without spaces, and it solved the problem!

I continued testing outside of Texworks and found that the quotation marks around C:\Program Files (x86)\TeXworks\texworks.exe were the reason the jump was working when using my own MIKTEX_EDITOR variable. Both "/" and "\\" separators seem to be accepted.
So ideally, to prevent the problem, I would suggest that TeXworks adds/overrides the MIKTEX_EDITOR variable with "" quotation marks added around the executable path, so that it works even when the path contains spaces.
